app/main.py

```diff
@@ -1,3 +1,71 @@
+import operator
+from typing import Union, Callable
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance") -> "Distance":
```

Comment:
Is the Distance the only type for other?
Please check all below method and make sure type hints everywhere are set according to the task requirements